controllers/operatorpolicy_controller.go

```diff
@@ -495,6 +526,143 @@ func buildSubscription(
 	return subscription, nil
 }
 
+func (r *OperatorPolicyReconciler) handleCSV(ctx context.Context,
+	policy *policyv1beta1.OperatorPolicy,
+	sub *operatorv1alpha1.Subscription,
+) (*operatorv1alpha1.ClusterServiceVersion, error) {
+	// case where subscription is nil
+	if sub == nil {
+		// need to remove all relatedobjects
+		err := r.removeObjsOfType(ctx, policy, clusterServiceVersionGVK)
+		if err != nil {
+			return nil, fmt.Errorf("error removing relatedObjects: %w", err)
+		}
+
+		err = r.updateStatus(ctx, policy, noDeploymentsCond, noExistingCSVObj)
+		if err != nil {
+			return nil, fmt.Errorf("error updating the status for Deployments: %w", err)
+		}
+
+		return nil, nil
+	}
+
+	watcher := opPolIdentifier(policy.Namespace, policy.Name)
+
+	// case where status has not been populated yet
+	if sub.Status.InstalledCSV == "" {
+		return nil, nil
+	}
```

Comment:
I think this should be handled the same as if the subscription was `nil`, it needs to kind of "reset" the condition and relatedObjects.